### PR TITLE
ifcfg: init at 0.23

### PR DESCRIPTION
**Pull request by movefasta**
###### Description of changes

Adds https://github.com/ftao/python-ifcfg. It is useful for pulling information such as IP, Netmask, MAC Address, Hostname, etc.

This is a re-creation of https://github.com/NixOS/nixpkgs/pull/164515 that's author is not active right now.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by azahi** [comment]
> Not sure about this but usully we add `pytestCheckHook` to `checkInputs` to enforce running tests the nixpkgs way.

`pytestCheckHook` doesn't work here.

File: pkgs/development/python-modules/ifcfg/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ftao";
+    repo = "python-ifcfg";
+    rev = "releases/${version}";
+    hash = "sha256-j621Yx/LjLaHPZ1N95cUSGxl0qESISS+HeQJRETLeoo=";
+  };
+
+  propagatedBuildInputs = [ iproute2 ];
+
```
Comment (azahi) [inline pkgs/development/python-modules/ifcfg/default.nix:22]: I think this can be removed.
```suggestion
```

**Comment by lopsided98** [comment]
The original author of #164515 responded today, so I think we should use that PR. In any case, I left some more review comments there, some of which apply to this PR as well.

**Comment by onny** [comment]
Thank you @movefasta for your work. Lets hope we can get this merged in the other PR

**Pull request closed without merging**